### PR TITLE
Graphui

### PR DESCRIPTION
**Pull request by ciliakas**
Feature updates:
- Added pie charts based on importance
- Added bar chart for last six month balance
- Added some ui for the charts
Code udpates:
- Added a few constants to Handler
- Added a few constants to GraphDrawer
- Added a find oldest entry month function to datafilter

### CODE REVIEW

File: SmartSaver/forms/reports/FormNetWorth.cs [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
     {
-        public FormNetWorth()
+        private readonly Handler _handler;
+        public FormNetWorth(Handler handler)
         {
             InitializeComponent();
+
+            _handler = handler;
+
+            ShowChart();
```
Comment (Stokalas) [inline SmartSaver/forms/reports/FormNetWorth.cs:19]: empty lines here seems redundant (minor thing)

File: SmartSaver/forms/reports/FormNetWorth.cs [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
+                if (balance > highestValue)
+                {
+                    highestValue = balance;
+                }
+                valueList.Add(balance);
+                namesList.Add(oldestMonth.ToShortDateString());
+                oldestMonth = TimeManager.MoveToNextMonth(oldestMonth);
+            }
+
+            var size = new Size(999,600);
```
Comment (Stokalas) [inline SmartSaver/forms/reports/FormNetWorth.cs:43]: Maybe these could be const variables?

File: SmartSaver/forms/reports/FormSpending.cs [hunk 3]
```diff
@@ -1,12 +1,85 @@
...
+
+        private void ShowSpending(PictureBox pictureBox)
+        {
+            var valuesList = new List<decimal>();
+            var namesList = new List<string>();
+
+            foreach (var importance in from importance in Handler.ImportanceList
+                                       let value = _handler.DataTotalsCalculator.GetTotaledExpenses(importance)
+                where value > 0
+                select importance)
```
Comment (Stokalas) [inline SmartSaver/forms/reports/FormSpending.cs:74]: Is it supposed to look like this or github butchering code again?

**Review by Stokalas — APPROVED** [review]
While everything seems to be fine I commented on some minor details

File: SmartSaver/forms/reports/FormSpending.cs (hunk 3 above)
Comment (MatasLa) [reply SmartSaver/forms/reports/FormSpending.cs:74]: I believe its github again.

**Review by MatasLa — APPROVED** [review]
Everything seems to work fine. Great job with graphs implementation.

File: SmartSaver/forms/reports/FormNetWorth.cs (hunk 1 above)
Comment (ciliakas) [reply SmartSaver/forms/reports/FormNetWorth.cs:20]: Just github being github, spacing looks fine in Visual Studio

File: SmartSaver/forms/reports/FormNetWorth.cs (hunk 2 above)
Comment (ciliakas) [reply SmartSaver/forms/reports/FormNetWorth.cs]: Good point, left this in here in a rush

File: SmartSaver/forms/reports/FormSpending.cs (hunk 3 above)
Comment (ciliakas) [reply SmartSaver/forms/reports/FormSpending.cs]: This is Visual Studio doing it's thing, tried to make this look more neat

**Review by Stokalas — APPROVED** [review]
Well done

**Pull request merged**